### PR TITLE
feat: python2 compatibility

### LINKED ISSUE #5: Python 2 compatibility
Hi.

Any chance for adding python 2 backwards compatibility?

I know python2 has been EOL officially now for a while, but it's still being supported by redhat and the ansible execution environment I'm  trying to use the collection in  (Redhat satellite). is stuck on ansible 2.9 and python 2.7 and I think Ansible Tower is faced with a similar issue.

### PR DESCRIPTION
**Pull request by gardar**
First attempt at python 2 backport

fixes #5

### CODE REVIEW

**Comment by maxhoesel** [comment]
This looks "good" to me (well, as good as backwards-compatibly Py2 code can look anyways, hah). The CI failures are unrelated - I've fixed them on the main branch.

As for merging this, I would like to have some more testing first. At the very least, the module tests should also be run against Python 2.7. The Modules workflow (in [ci.yml](https://github.com/maxhoesel/ansible-collection-caddy/blob/039bd498aa4e008455c5d06ac163ff40cbafc51c/.github/workflows/ci.yml#L50)) already tests against Python 3.6, we could just extend it with a strategy matrix and also add python 2.7.

Oh, and the docs would need to reflect this change as well of course

**Comment by gardar** [comment]
@maxhoesel  Updated testing to include 2.7 in addition to 3.6 and updated docs, hope you can merge soon.

**Review by maxhoesel — COMMENTED** [review]
Had another look at it, LGTM! Some minor things here and there but I'll try to get this merged soon - then I'll have a shot at backporting the new create_path feature to py27

File: plugins/modules/caddy_config.py [hunk 1]
```diff
@@ -124,7 +124,7 @@ def delete_config(module: AnsibleModule, server: CaddyServer):
             # The object doesn't exist, nothing to delete
             return {"changed": False}
         else:
-            module.fail_json(msg=f"Error while retrieving current configuration: {current_config_or_error['error']}")
+            module.fail_json(msg="Error while retrieving current configuration: {current_config_or_error['error']}".format(**locals()))
```
[the remark this answers is not in the document]
Comment (gardar) [reply plugins/modules/caddy_config.py:127]: No I left this there by accident, at first I tested locally against python 2 by replacing all the f-strings with **locals() just to be able to catch everything without having to be specific... Then I replaced it with the specific values.

I think it's controversial, where some prefer using locals() while others don't.  What's your opinion?

File: README.md [hunk 2]
```diff
@@ -28,7 +28,7 @@ An Ansible collection containing roles/modules to install, configure and interac
 
 - A recent release of Ansible. This collection officially supports the 2 most recent Ansible releases.
   Older versions might still work, but are not supported
-- Python 3.6 or newer
+- Python 2.7 or newer
```
[the remark this answers is not in the document]
Comment (gardar) [reply README.md]: Good point, fixed!

**Pull request merged**